src/main/xerus/monstercat/api/Player.kt

```diff
@@ -138,32 +138,38 @@ object Player: FadingHBox(true, targetHeight = 25) {
 	/** Plays the given [track] in the Player, stopping the previous MediaPlayer if necessary */
 	fun playTrack(track: Track) {
 		disposePlayer()
-		val hash = track.streamHash ?: run {
+		if(!track.streamable) {
 			showError("$track is currently not available for streaming!")
 			return
 		}
-		updateCover(track.release.coverUrl)
-		logger.debug("Loading $track from $hash")
-		activePlayer.value = MediaPlayer(Media("https://s3.amazonaws.com/data.monstercat.com/blobs/$hash"))
-		updateVolume()
-		onFx {
-			activeTrack.value = track
-		}
-		playing("Loading $track")
-		player?.run {
-			play()
-			setOnEndOfMedia { playNext() }
-			setOnReady {
-				label.text = "Now Playing: $track"
-				val total = totalDuration.toMillis()
-				seekBar.progressProperty().dependOn(currentTimeProperty()) { it.toMillis() / total }
-				seekBar.transitionToHeight(Settings.PLAYERSEEKBARHEIGHT(), 1.0)
+		var streamUrl: String? = null
+		GlobalScope.launch {
+			streamUrl = APIConnection.getRedirectedStreamURL(track)
+		}.invokeOnCompletion {
```

Comment:
why `invokeOnCompletion` instead of just putting the code directly after it? If I'm not mistaken this line could just be removed?!

Reply:
Otherwise the code will get executed before streamUrl is set, not playing the track. This waits until the streamURL is fetched (because it's an HTTP request it's not synchronous and instant) to load the track.

Reply:
but if you removed this single line, all the other code would still be executed after the streamUrl is set, no?

Reply:
I get it. yeah we could dump invokeOnCompletion without changing how it works in the end.

Reply:
c678932be460ba0daf0f615f534a9588823a9d48 works fine